include/boost/thread/pthread/condition_variable.hpp

```diff
@@ -274,10 +306,9 @@ namespace boost
                 const chrono::time_point<Clock, Duration>& t)
         {
           using namespace chrono;
-          system_clock::time_point     s_now = system_clock::now();
+          internal_clock_t::time_point s_now = internal_clock_t::now();
           typename Clock::time_point  c_now = Clock::now();
-          wait_until(lock, s_now + ceil<nanoseconds>(t - c_now));
-          return Clock::now() < t ? cv_status::no_timeout : cv_status::timeout;
+          return wait_until(lock, s_now + ceil<nanoseconds>(t - c_now));
```

Comment:
I believe that we should iterate until  Clock::now() >= t

Reply:
See my comments above.